src/lib/api/core/renderer-adapter.ts

```diff
@@ -0,0 +1,70 @@
+import {Renderer, Renderer2, RendererStyleFlags2} from '@angular/core';
+
+/**
+ * Adapts the 'deprecated' Angular Renderer v1 API to use the new Renderer2 instance
+ * This is required for older versions of NgStyle and NgClass: which require
+ * the v1 API (but should use the v2 instances)
+ */
+export class RendererAdapter extends Renderer {
+  constructor(private _renderer: Renderer2) {
+    super();
+  }
+
+  setElementClass(el: any, className: string, isAdd: boolean): void {
+    if (isAdd) {
+      this._renderer.addClass(el, className);
+    } else {
+      this._renderer.removeClass(el, className);
+    }
+  }
+
+  setElementStyle(el: any, styleName: string, styleValue: string): void {
+    if (styleValue) {
+      this._renderer.setStyle(el, styleName, styleValue);
+    } else {
+      this._renderer.removeStyle(el, styleName);
+    }
+  }
+
+  // new API is forwarded
+  addClass(el: any, name: string): void {
+    this._renderer.addClass(el, name);
+  }
+
+  removeClass(el: any, name: string): void {
+    this._renderer.removeClass(el, name);
+  }
+
+  setStyle(el: any, style: string, value: any, flags?: RendererStyleFlags2): void {
+    this._renderer.setStyle(el, style, value, flags);
+  }
+
+  removeStyle(el: any, style: string, flags?: RendererStyleFlags2): void {
+    this._renderer.removeStyle(el, style, flags);
+  }
+
+  // ******************************************************************
+  // !! Renderer is an abstract class with abstract methods
+  //
+  // These are implementation of those methods... and do NOTHING since
+  // we only use setElementStyle() and setElementClass()
+  // ******************************************************************
+
+  selectRootElement(): any { }
```

Comment:
If these methods aren't meant to be used, consider throwing an error or removing the implementation so they fall back to the inherited ones from the `Renderer`?

Reply:
We cannot remove the implementation... We want to to appear as a Renderer1 instance but built from a Renderer2. Renderer is an abstract class with many abstract methods. This is we have to implement them EVEN if ignored. 

But we can throw error as not-implemented!

Fixed.